dds/DCPS/DiscoveryBase.h

```diff
@@ -1247,40 +1271,56 @@ namespace OpenDDS {
 
         MatchingData md;
 
-        // if the type object is not in cache, send RPC request
-        md.writer = writer;
-        md.reader = reader;
+        // If the type object is not in cache, send RPC request
         md.time_added_to_map = MonotonicTimePoint::now();
 
+        // NOTE(sonndinh): Is it possible for a discovered endpoint to include only the "complete"
+        // part in its TypeInformation? If it's possible, then we may need to handle that case, i.e.,
+        // request only the remote complete TypeObject (if it's not already in the cache).
+        // The following code assumes when the "minimal" part is not included in the discovered
+        // endpoint's TypeInformation, then the "complete" part also is not included.
```

Comment:
Do we expect to resolve before merging this PR?

Reply:
No, I think it's not necessary to have this resolve in this PR. This is more of a note on whether our current assumption, that the minimal member is always present in `TypeInformation`, is compatible with the spec. If we decide that we need to make some changes to this part, I'd like to do it in a separate PR.

Reply:
If our goal for this task is to only have self compatibility up and running I don't think it needs to be resolved, but this note will be helpful for if/when we worry about interoperability in the future.

Reply:
If the remote peer provides only a Complete TypeIdentifier in discovery, we could request the corresponding Complete TypeObject via RPC and then run the complete-to-minimal algorithm on it.